src/prefect/cli/agent.py

```diff
@@ -274,6 +274,13 @@ def kubernetes():
     "values can be provided as a comma-separated list "
     "(e.g. `--image-pull-secrets VAL1,VAL2`)",
 )
+@click.option(
+    "--disable-job-deletion",
+    "delete_finished_jobs",
+    help="Turn off automatic deletion of finished jobs in the namespace.",
+    is_flag=True,
+    default=True,
+)
```

Comment:
> Hey @joshmeek 😜
> 
> This seems like it'd be nicely paired with a TTL https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/ option. Perhaps we'd be better off just adding a `--ttl-after-finished <seconds: int>` option defaulting to 0 which retains the default behavior of immediate deletion but adds the TTL annotation otherwise?
> 
> cc @mashun4ek for the implications of it being a beta feature

Hello @joshmeek!!! :) 
@madkinsz Thank you for pointing to TTL controller, just find out about it. The only implication I see that this feature will be utilized in clusters running >=1.21 version of Kubernetes. I wonder if we can use this https://kubernetes.io/docs/concepts/workloads/controllers/job/#ttl-mechanism-for-finished-jobs

Reply:
Hello @joshmeek!!! :)
@madkinsz Thank you for pointing to TTL controller, just find out about it. The only implication I see that this feature will be utilized in clusters running >=1.21 version of Kubernetes, and currently most cloud providers offer 1.18-1.19 version of Kubernetes for managed cluster.

Reply:
Hey @madkinsz and @mashun4ek 🙂  hope all is well!

Reply:
My docker engine thinks it's 1970 but otherwise it's going great :D